### ISSUE TITLE
Treeshaking not working

### ISSUE THREAD

**Issue by ekwoka** [graywolftech/react-heroicons#11]
The way this package is bundled is not able to be treeshaken.

It uses a single exported object, which means all components are included.

If these were changed to separate exports, they could remove all the dead code.

**Comment by jsmith** [comment]
Hey @ekwoka I just looked at the output `index.js` and there is a single export but it exports each component individually (not a single object I am pretty sure). Was treeshaking not working for you?

```
export { AcademicCapOutline, AcademicCapSolid, ..., ZoomOutOutline, ZoomOutSolid };
```

**Comment by ekwoka** [comment]
It was not.

The package was not marked sideEffect free, so many bundlers are hesitant to attempt tree shaking if they can't easily tell.

So vite and esbuild, for example.

If you import and use just a single one, it adds the entire collection to the bundle (that's over 600 with the new ones).

Luckily I fixed it in the PR :)

**Comment by jsmith** [comment]
Wow I had no idea about `sideEffect`, thanks so much!

**Comment by ekwoka** [comment]
This is quite interesting. With the new one, it still doesn't seem to work.

Just doing:

```ts
import { PlusSmallSolid } from '@graywolfai/react-heroicons';

console.log(PlusSmallSolid);
```

and bundling with esbuild includes all the icons (unminified: 747kb) and react stuff

I published my own fork of this for preact at `preact-heroicons`

Doing the same thing:

```ts
import { PlusSmallSolid } from 'preact-heroicons';

console.log(PlusSmallSolid);
```

It's just the one icon and preact stuff (unminified: 24.1kb)

The only real difference is I don't have them all in one file (just barreled like the prior versions of your library). I'm not sure if other systems do different treeshaking (like maybe esbuild does it by imports from files and not any treeshaking within a single file?)


ah!

Found the issue!

if you place `/* @__PURE__ */` before the `React.forwardRef(` in each icon it will be treeshaken properly.

The `sideEffects` in the package.json applies to the entire file, saying that each file is sideEffect free when it's not used at all, but does not apply to each individual export.

So I guess the options are separate files with barrel, or adding `/* @__PURE__ */` before the call to forwardRef. I guess since it does a function call and isn't just a function itself, it can't be sure that that function isn't doing other things.

**Comment by jsmith** [comment]
Wow thank you so much for looking into this! I really should have confirmed tree shaking worked before pushing the latest version of the library. I just released v2.0.1 and confirmed that tree shaking is finally working, at least with esbuild :)
